[NNC] Add unroll and flatten APIs which not require return stmt pointer

huiguoo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#56420 [NNC] Add unroll and flatten APIs which not require return stmt pointer**

Differential Revision: [D27866118](https://our.internmc.facebook.com/intern/diff/D27866118)

huiguoo committed:
[NNC] Add unroll and flatten APIs which not require return stmt pointer

[ghstack-poisoned]

navahgar commented on torch/csrc/jit/tensorexpr/loopnest.cpp:
```diff
@@ -1658,6 +1658,11 @@ void LoopNest::unroll(For* f, Stmt** unrolled) {
 }
 
+void LoopNest::unroll(For* f) {
+  Stmt* unrolled;
```
Can you initialize this to `nullptr` so that clang-tidy doesn't complain?

huiguoo replied:
Oh, right. For consistency, I copied what `sliceHead` does and added an annotation to prevent the complains :)

navahgar commented on torch/csrc/jit/tensorexpr/loopnest.cpp:
```diff
@@ -1764,6 +1769,11 @@ bool LoopNest::flatten(const std::vector<For*>& loops, For** flattened) {
 }
 
+bool LoopNest::flatten(const std::vector<For*>& loops) {
+  For* flattened;
```
Again, initialize to `nullptr`.

navahgar reviewed (approved):
LGTM

huiguoo committed:
Update on "[NNC] Add unroll and flatten APIs which not require return stmt pointer"


Differential Revision: [D27866118](https://our.internmc.facebook.com/intern/diff/D27866118)

[ghstack-poisoned]

huiguoo committed:
Update on "[NNC] Add unroll and flatten APIs which not require return stmt pointer"


Differential Revision: [D27866118](https://our.internmc.facebook.com/intern/diff/D27866118)

[ghstack-poisoned]

huiguoo committed:
Update on "[NNC] Add unroll and flatten APIs which not require return stmt pointer"


Differential Revision: [D27866118](https://our.internmc.facebook.com/intern/diff/D27866118)

[ghstack-poisoned]
